nexus/src/app/instance_network.rs

```diff
@@ -230,175 +227,6 @@ pub(crate) async fn boundary_switches(
-    //    the new instance record.
-    // 5. The instance migrates from T to T' and Nexus applies networking
-    //    configuration reflecting that the instance is on T'.
-    // 6. The update in step 2 applies configuration saying the instance
-    //    is on sled T.
-    if new_instance_state.migration_id.is_some() {
-        debug!(log,
-                "instance still has a migration in progress, won't touch \
-                network config";
-                "instance_id" => %instance_id,
-                "migration_id" => ?new_instance_state.migration_id);
-
-        return Ok(());
-    }
-
-    let new_propolis_id = new_instance_state.propolis_id.unwrap();
-
-    // Updates that end live migration need to push OPTE V2P state even if
-    // the instance's active sled did not change (see below).
-    let migration_retired = prev_instance_state.migration_id.is_some()
-        && new_instance_state.migration_id.is_none();
-
-    if (prev_instance_state.propolis_id
-        == new_instance_state.propolis_id.map(GenericUuid::into_untyped_uuid))
-        && !migration_retired
-    {
-        debug!(log, "instance didn't move, won't touch network config";
-                "instance_id" => %instance_id);
-
-        return Ok(());
-    }
-
-    // Either the instance moved from one sled to another, or it attempted
-    // to migrate and failed. Ensure the correct networking configuration
-    // exists for its current home.
-    //
-    // TODO(#3107) This is necessary even if the instance didn't move,
-    // because registering a migration target on a sled creates OPTE ports
-    // for its VNICs, and that creates new V2P mappings on that sled that
-    // place the relevant virtual IPs on the local sled. Once OPTE stops
-    // creating these mappings, this path only needs to be taken if an
-    // instance has changed sleds.
-    let new_sled_id = match datastore
-        .vmm_fetch(&opctx, authz_instance, &new_propolis_id)
-        .await
-    {
-        Ok(vmm) => vmm.sled_id,
-
-        // A VMM in the active position should never be destroyed. If the
-        // sled sending this message is the owner of the instance's last
-        // active VMM and is destroying it, it should also have retired that
-        // VMM.
-        Err(Error::ObjectNotFound { .. }) => {
-            error!(log, "instance's active vmm unexpectedly not found";
-                    "instance_id" => %instance_id,
-                    "propolis_id" => %new_propolis_id);
-
-            return Ok(());
-        }
-
-        Err(e) => return Err(e),
-    };
-
-    v2p_manager.activate();
-
-    let (.., sled) =
-        LookupPath::new(opctx, datastore).sled_id(new_sled_id).fetch().await?;
-
-    instance_ensure_dpd_config(
-        datastore,
-        log,
-        resolver,
-        opctx,
-        opctx_alloc,
-        instance_id,
-        &sled.address(),
-        None,
-    )
-    .await?;
-
-    Ok(())
-}
```

Comment:
Network config updates are now performed by the `instance-update` saga. This function is basically subsumed by it, so it's gone away.